src/ast/nodes.hpp

```diff
@@ -62,7 +70,24 @@ struct Function
   CodeBlock body;
 };
 
-using TranslationUnit = std::vector<Function>;
+using Method = Function;
+
+struct AccessSpecifier
+{
+  LevelSpecifier level;
+};
+
+struct Class
+{
+  std::string_view name;
+  std::vector<std::pair<Attribute, AccessSpecifier>> attributes;
+  std::vector<std::pair<Method, AccessSpecifier>> methods;
```

Comment:
I'm definetly overthinking the performance aspect of the compiler, but I feel like at some points we'll need to loop over all public attributes in a class. Separating attributes & methods in 3 different vectors will avoid some small amount of overhead.
Regardless, let's ignore my comment, because we can just change this later on if it makes sense